### ISSUE TITLE
Changing a filter operator with many values, only keep first value

### ISSUE THREAD

**Issue by rephus** [lightdash/lightdash#3582]
### Description

[Screencast from 26-10-22 09:00:13.webm](https://user-images.githubusercontent.com/1983672/197956925-59afae58-3f78-4b47-98a1-ebe5323d7f48.webm)

This is because on filters we have values as any[], but when changing the filter, we only take the first `values[0]` Last time I checked, a bit of refactoring was needed to fix this.

**Comment by TuringLovesDeathMetal** [comment]
Is this related to this issue? https://github.com/lightdash/lightdash/issues/3085

**Comment by rephus** [comment]
Yes, duplicated of https://github.com/lightdash/lightdash/issues/3085 :facepalm:
